src/k8s/pkg/k8sd/types/cluster_config_validate.go

```diff
@@ -20,6 +21,24 @@ func validateCIDRs(cidrString string) error {
 	return nil
 }
 
+// isValidAddress checks if a given address is valid. Allows with and without schema.
+func isValidAddress(address string) bool {
+	if u, err := url.Parse(address); err == nil && u.Host != "" {
+		return true
+	}
+
+	host, port, err := net.SplitHostPort(address)
+	if err == nil {
+		if host == "localhost" || net.ParseIP(host) != nil {
```

Comment:
ah, good point. any valid hostname should be allowed. if `url.Parse` succeeded above, then we should be good, extra checks here are probably not needed.

let's add unit tests that verify all these

Reply:
done